reflex/utils/format.py

```diff
@@ -598,7 +600,14 @@ def format_state(value: Any) -> Any:
     """
     # Handle dicts.
     if isinstance(value, dict):
-        return {k: format_state(v) for k, v in value.items()}
+        result = {}
+        for k, v in value.items():
+            if isinstance(v, (pathlib.PosixPath, functools.partial)):
```

Comment:
I think we can be more general here by either:
1. Using `reflex.utils.serailizers.has_serializer` to check
2. Just putting a try/except around the setting.

Actually I think the easiest fix would be to pass down the key to `format_state` and just update the error message thrown there.

Reply:
Thanks, @picklelo, your suggestion makes sense.

Reply:
I have now updated the code by passing the key to `format_state` and updating the error message